File: src/main/java/factory/DualWeekScheduleViewFactory.java
package factory;

import app.FXMLPaths;
import control.schedule.DualWeekScheduleViewController;
import control.schedule.WeekScheduleController;
import javafx.scene.Node;

import java.io.IOException;

public class DualWeekScheduleViewFactory<C extends DualWeekScheduleViewController<C1,C2>, C1 extends WeekScheduleController, C2 extends WeekScheduleController>
    extends ViewFactory<C>{

    public DualWeekScheduleViewFactory(C controller){
        super(FXMLPaths.DualWeekGenericSchedule());
        this.controller = controller;
    }

    @Override
    public Node load() throws IOException {
        Node n1 = new ScheduleViewFactory<>(controller.getFirstWeekController()).load();
        Node n2 = new ScheduleViewFactory<>(controller.getSecondWeekController()).load();

        Node n = super.load(controller); //main controller has to be loaded after the content of both weeks has been instantiated

        controller.setAWeekContent(n1);
        controller.setBWeekContent(n2);

        return n;
    }

    @Override
    public Node load(C controller) throws IOException {
        throw new UnsupportedOperationException();
    }
}

File: src/main/scala/factory/DualWeekScheduleViewFactory.scala
package factory

import java.io.IOException

import app.FXMLPaths
import control.schedule.{DualWeekScheduleViewController, WeekScheduleController}

import javafx.scene.Node

class DualWeekScheduleViewFactory[C <: DualWeekScheduleViewController[C1, C2],
    C1 <: WeekScheduleController,
    C2 <: WeekScheduleController] (controller: C) extends ViewFactory[C](FXMLPaths.DualWeekGenericSchedule) {

    @throws[IOException]
    override def load: Node = {
        val n1 = new ScheduleViewFactory[C1](controller.getFirstWeekController).load
        val n2 = new ScheduleViewFactory[C2](controller.getSecondWeekController).load

        //main controller has to be loaded after the content of both weeks has been instantiated
        val n = super.load(controller)

        controller.setAWeekContent(n1)
        controller.setBWeekContent(n2)

        n
    }

    @throws[UnsupportedOperationException]
    override def load(controller: C) = throw new UnsupportedOperationException
}
